src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt

```diff
@@ -191,18 +321,73 @@ class MangaPlusCreators(override val lang: String) : HttpSource() {
         return GET(page.imageUrl!!, newHeaders)
     }
 
-    private fun Response.asMpcResponse(): MpcResponse = use {
-        json.decodeFromString(body.string())
-    }
-
     companion object {
-        private const val API_URL = "https://medibang.com/api/mpc"
+        private val CHAPTER_DATE_FORMAT by lazy {
+            SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
+        }
+        private const val BASE_URL = "https://mangaplus-creators.jp"
+        private const val API_URL = "$BASE_URL/api"
         private const val USER_AGENT = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) " +
             "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36"
+        const val PREFIX_TITLE_ID_SEARCH = "title:"
+        const val PREFIX_EPISODE_ID_SEARCH = "episode:"
+        const val PREFIX_AUTHOR_ID_SEARCH = "author:"
+    }
 
-        private const val POPULAR_PAGE_SIZE = "30"
-        private const val CHAPTER_PAGE_SIZE = "200"
+    private fun parseChapterDate(dateStr: String): Long {
+        return runCatching { CHAPTER_DATE_FORMAT.parse(dateStr)?.time }
+            .getOrNull() ?: 0L
+    }
```

Comment:
use helper from core lib
`keiyoushi.utils.tryParse`

```kt
dateFormat.tryParse(date)
```

Reply:
was gonna suggest adding that to contributing.md, but it looks like steven wrote it right after I had gone through it past week XD

Reply:
seems to be duplicated in multiple places similar to https://github.com/keiyoushi/extensions-source/pull/10337#discussion_r2312918285